src/toil/job.py

```diff
@@ -1483,6 +1483,9 @@ def __init__(self, job):
         """
         # Giving the root of the subgraph the same resources as the first job in the subgraph.
         Job.__init__(self, **job._requirements)
+        if job._directPredecessors:
+            for job_ in job._directPredecessors:
+                job_.addChild(self)
         self.encapsulatedJob = job
```

Comment:
Excellent - this looks like the right thing to do. I would add a comment above #1486 to say:

Ensure that the encapsulated job has the same direct predecessors as the job being encapsulated.